### ISSUE TITLE
Empty file TypeError: Cannot read property 'getFormat' of undefined

### ISSUE THREAD

**Issue by ChaosySama** [kucherenko/jscpd#379]
**Describe the bug**
When the detected file is empty, the following error appears:
```
TypeError: Cannot read property 'getFormat' of undefined
```
and the subsequent detects are interrupted.

I use new InFilesDetector(), and option.mode = weak

**Expected behavior**
Skip these empty file or not interrupt the subsequent detects

**Desktop (please complete the following information):**
 - OS: MacOS
 - OS Version 10.13
 - NodeJS Version 10
 - jscpd version 3.3.0-rc.10

**Comment by kucherenko** [comment]
Thank you for the report, will investigate

**Comment by charrismatic** [comment]
Doing some investigating here to see if I can get it work. 

Error:

```js
TypeError: Cannot read property 'getFormat' of undefined
    at Statistic.matchSource (/home/user/lib/node_modules/jscpd/node_modules/@jscpd/core/dist/statistic.js:50:31)
    at Detector.emit (/home/user/lib/node_modules/jscpd/node_modules/eventemitter3/index.js:181:35)
    at Detector.<anonymous> (/home/user/lib/node_modules/jscpd/node_modules/@jscpd/core/dist/detector.js:37:22)
    at Generator.next (<anonymous>)
    at /home/user/lib/node_modules/jscpd/node_modules/@jscpd/core/dist/detector.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/home/user/lib/node_modules/jscpd/node_modules/@jscpd/core/dist/detector.js:4:12)
    at detect (/home/user/lib/node_modules/jscpd/node_modules/@jscpd/core/dist/detector.js:36:50)
    at Detector.<anonymous> (/home/user/lib/node_modules/jscpd/node_modules/@jscpd/core/dist/detector.js:51:20)
    at Generator.next (<anonymous>)
```

In the file `/home/user/lib/node_modules/jscpd/node_modules/@jscpd/core/dist/statistic.js` 

I threw  a log statement at `matchSource` before it tries to `getFormat`

```js
    matchSource(payload) {
        console.log(payload);
        const { source } = payload;
        const format = source.getFormat() || 'javascript';
        if (!(format in this.statistic.formats)) {
```

Right before the error throws the log outputs an empty payload.

```js
...
{ source: undefined }
```

I'll let you know if I'm able to get it fixed in the meantime

**Comment by charrismatic** [comment]
The error is caused by a blank file containing several empty lines. This block coincidentally matched another file with the same block of empty lines. Im guessing that the detector sees this as a match but the tokenizer returns null or undefined. After finding this file and removing all the blank lines (without deleting the file) the command works again.

Kind of a neat edge case so im not sure if you want to try and handle this exception, but the error can be corrected by fixing adjusting the files in the code base (if that is an option)

**Comment by kucherenko** [comment]
Thank you, will investigate and fix

**Comment by kucherenko** [comment]
fixed at jscpd@3.3.1
